engine/graphics/proto/graphics/graphics_ddf.proto

```diff
@@ -192,10 +195,11 @@ message ShaderDesc
 
     message Shader
     {
-        required Language        language   = 1;
-        optional bytes           source     = 2;
-        repeated ResourceBinding uniforms   = 4;
-        repeated ResourceBinding attributes = 5;
+        required Language        language              = 1;
+        optional bytes           source                = 2;
+        repeated ResourceBinding uniforms              = 4;
+        repeated ResourceBinding attributes            = 5;
+        optional bool            variant_texture_array = 6;
```

Comment:
Right now this is the only variant we have so we don't need anything fancy until we need more variants